frontends/ol-network-interface/src/index.ts

```diff
@@ -0,0 +1,91 @@
+export type NetworkHeaders = Record<string, string>
+
+type Method = "GET" | "POST" | "PATCH" | "DELETE"
+
+type JSONBody = Record<string, unknown> | unknown[] | null | undefined
+
+interface NetworkResponse<T = unknown> {
+  status: number
+  body: T
+}
+
+interface NetworkInterface {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  makeRequest: <T = unknown>(
+    method: Method,
+    url: string,
+    body?: JSONBody,
+    headers?: NetworkHeaders
+  ) => Promise<NetworkResponse<T>>
+  get: <T = unknown>(
+    url: string,
+    headers?: NetworkHeaders
+  ) => Promise<NetworkResponse<T>>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  post: <T = unknown>(
+    url: string,
+    body?: JSONBody,
+    headers?: NetworkHeaders
+  ) => Promise<NetworkResponse<T>>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  patch: <T = unknown>(
+    url: string,
+    body?: JSONBody,
+    headers?: NetworkHeaders
+  ) => Promise<NetworkResponse<T>>
+  delete: <T = unknown>(
+    url: string,
+    headers?: NetworkHeaders
+  ) => Promise<NetworkResponse<T>>
+}
+
+class NetworkError extends Error {
+  status: number
+
+  constructor(status: number, msg: string) {
+    super(msg)
+    this.status = status
+  }
+}
+
+const makeRequest: NetworkInterface["makeRequest"] = async (
```

Comment:
This implements the `NetworkInterface` above with native fetch.

I have not, for the moment, included any tests for this. I do think we should use `react-query`, and I do think we should interact with it through some standard adapter like this `NetworkInterface`. I am a bit torn about whether
- we should implement network interface with library like axios or superagent
- or keep this minimal fetch wrapper.

I lean toward keeping the minimal fetch implementation: axios and superagent both have a fair amount of stuff we don't need. (Some of the "stuff we don't need" is because react-query manages for us, e.g., request retries.) *Open-discussions is already using superagent via `redux-query-interface-superagent`. But redux-query-interface-superagent specifies superagent v5, which is deprecated: https://www.npmjs.com/package/superagent/v/5.3.1*

Anyway, if people are ok keeping this fetch implementation, i could add simple tests for it 🤷